docs: basic lineage release notes (#9708)
determined-ai/determined: docs/release-notes/basic-lineage.rst

:orphan:

**New Features**

-  WebUI: View Pachyderm data lineage

>    -  Experiments with a Pachyderm data integration configured will now show a link to the Pachyderm

Reviewer: ```suggestion
- Experiments with configured Pachyderm data integration now display a link to the Pachyderm
```

>       repo in the trial view page. The link to the Pachyderm repo will also be available when

Reviewer: ```suggestion
repo in the trial view page. The link is also available when
```
